gcc/rust/ast/rust-ast-dump.cc

```diff
@@ -364,7 +366,10 @@ Dump::visit (AsyncBlockExpr &expr)
 // rust-item.h
 void
 Dump::visit (TypeParam &param)
-{}
+{
+  // Is it possible to have a null type here?
+  param.get_type ()->accept_vis (*this);
```

Comment:
I need to double-check but off the top of my head there is a case it can be null but it might be a different data structure. I am thinking about the case of:



```
impl<T=i32> Foo<T> {}
```

Reply:
I think that is a different data structure yeah. I'll make sure before merging this

Reply:
Yep, this counts as an inherent impl and is covered by #1295